Error: Debug Failure. False expression.

## Current behavior

```
➜  yarn build
yarn run v1.22.10
$ webpack
Error: Debug Failure. False expression.
Error: Debug Failure. False expression.
    at Object.emitBuildInfo (/Users/ralex/code/web/node_modules/typescript/lib/typescript.js:107297:22)
    at emitTsBuildInfoFileForBuilderProgram (/Users/ralex/code/web/packages/components/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:85:25)
    at /Users/ralex/code/web/packages/components/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:281:37
    at Object.afterProgramCreate (/Users/ralex/code/web/packages/components/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/ControlledWatchCompilerHost.js:15:17)
    at synchronizeProgram (/Users/ralex/code/web/node_modules/typescript/lib/typescript.js:112808:22)
    at Object.createWatchProgram (/Users/ralex/code/web/node_modules/typescript/lib/typescript.js:112746:9)
    at Object.<anonymous> (/Users/ralex/code/web/packages/components/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:290:59)
    at Generator.next (<anonymous>)
    at /Users/ralex/code/web/packages/components/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:8:71
    at new Promise (<anonymous>)
Hash: e227d30820c3b2d629b8
Version: webpack 4.46.0
Time: 47698ms
Built at: 05/08/2021 5:04:29 PM
   Asset      Size  Chunks             Chunk Names
build.js  2.52 MiB    main  [emitted]  main
...elided...
```

I was able to work around this with the following config:

```js
    new ForkTsCheckerWebpackPlugin({
      typescript: {
        configOverwrite: {
          compilerOptions: {
            outFile: null
          }
        }
      }
    })
```

## Expected behavior
<!-- Describe what the desired behavior would be. -->
It should not error. 

Probably, it should instead silently merge `outFile: null` and perhaps also `outDir: null`…

## Steps to reproduce the issue
<!-- Describe how we can reproduce this problem step by step. -->
The following in a tsconfig may be sufficient, I'm not sure: 
```
"declaration": true,
"noEmit": false,
"emitDeclarationOnly": true,
"outFile": "./dist/index.d.ts",
```

## Issue reproduction repository
<!-- Location of a (minimal) repository that demonstrates the issue. -->

## Environment
 - **fork-ts-checker-webpack-plugin**: `6.2.6`
 - **typescript**: `4.2.4`
 - **eslint**: `6.8.x`
 - **webpack**: `4.46.0`
 - **os**: macOS

That command shows these errors:
```
[2:52:51 PM] Starting compilation in watch mode...

tsconfig.json:6:5 - error TS5053: Option 'emitDeclarationOnly' cannot be specified with option 'noEmit'.

6     "noEmit": false,
      ~~~~~~~~

tsconfig.json:7:5 - error TS5053: Option 'emitDeclarationOnly' cannot be specified with option 'noEmit'.

7     "emitDeclarationOnly": true,
      ~~~~~~~~~~~~~~~~~~~~~

[2:53:10 PM] Found 2 errors. Watching for file changes.
```
Without `--noEmit`, the command succeeds.

--- a/src/typescript-reporter/reporter/TypeScriptReporter.ts
+++ b/src/typescript-reporter/reporter/TypeScriptReporter.ts
@@ -100,7 +100,8 @@ function createTypeScriptReporter(configuration: TypeScriptReporterConfiguration
     if (
       configuration.mode !== 'readonly' &&
       parsedConfiguration &&
-      parsedConfiguration.options.incremental
+      parsedConfiguration.options.incremental &&
+      !parsedConfiguration.options.outFile
     ) {
       const program = builderProgram.getProgram();
       // eslint-disable-next-line @typescript-eslint/no-explicit-any

